stats/release_queue.php

```diff
@@ -291,75 +231,34 @@ function _show_queue_details($round, $name, $unheld_only)
-    dpsql_dump_themed_query("
-        SELECT
-
-            concat('$comments_url1',projectID,'$comments_url2', nameofwork, '$comments_url3') as '"
-                . DPDatabase::escape(_("Title")) . "',
-            authorsname as '" . DPDatabase::escape(_("Author")) . "',
-            language    as '" . DPDatabase::escape(_("Language")) . "',
-            genre       as '" . DPDatabase::escape(_("Genre")) . "',
-            difficulty  as '" . DPDatabase::escape(_("Difficulty")) . "',
-            username    as '" . DPDatabase::escape(_("Project Manager")) . "',
-            FROM_UNIXTIME(modifieddate) as '"
-                . DPDatabase::escape(_("Date Last Modified")) . "',
-            IF(ISNULL(project_holds.state),'&nbsp;','Y') AS '" . DPDatabase::escape(_("Hold?")) . "'
-        FROM projects
-            LEFT OUTER JOIN project_holds USING (projectid, state)
-        WHERE ($cooked_project_selector)
-            AND state='{$round->project_waiting_state}'
-            AND $unheld_only_sql
-        ORDER BY modifieddate, nameofwork
-    ");
-}
-
-function _get_num_projects_and_pages_released_in_last($name, $seconds_ago, $waiting_state)
-{
-    $sql = sprintf(
-        "
-        SELECT count(*)
-        FROM project_events
-        WHERE event_type = 'transition'
-            AND details1 = '%s'
-            AND details3 = 'via_q: %s'
-            AND timestamp >= %d
-        ",
-        DPDatabase::escape($waiting_state),
-        DPDatabase::escape($name),
-        $seconds_ago
-    );
-    $result = DPDatabase::query($sql);
-    $projects_released = mysqli_fetch_row($result)[0];
-
-    $sql = sprintf(
-        "
-        SELECT sum(n_pages)
-        FROM projects
-        WHERE projectid in (
-            SELECT projectid
-            FROM project_events
-            WHERE event_type = 'transition'
-                AND details1 = '%s'
-                AND details3 = 'via_q: %s'
-                AND timestamp >= %d
-        )
-        ",
-        DPDatabase::escape($waiting_state),
-        DPDatabase::escape($name),
-        $seconds_ago
-    );
-    $result = DPDatabase::query($sql);
-    $pages_released = mysqli_fetch_row($result)[0] ?? 0;
-
-    return [$projects_released, $pages_released];
+    $headers = [
+        _("Title"),
+        _("Author"),
+        _("Language"),
+        _("Genre"),
+        _("Difficulty"),
+        _("Project Manager"),
+        _("Date Last Modified"),
+        _("Hold?"),
+    ];
+    echo "<table class='themed theme_striped stats' style='width: 100%; table-layout: fixed'>\n";
+    echo "<tr>\n";
+    echo surround_and_join(array_map("html_safe", $headers), "<th>", "</th>\n", "");
+    echo "</tr>\n";
+    foreach(fetch_queue_projects_data($round, $queue["project_selector"], $unheld_only) as $p) {
+        echo "<tr>\n";
+        echo "<td><a href='$code_url/project.php?id=", html_safe($p["project_id"]), "'>", $p["title"], "</a></td>\n";
```

Comment:
`$p["title"]` needs `html_safe()` and `html_safe($p["project_id"])` should use `attr_safe()` instead although honestly I'd just output it directly since we control those and know they don't have anything special in them.

Reply:
<shakes head> It's so easy to miss these! Thanks.